code/code/misc/guild.cc

```diff
@@ -1174,7 +1174,7 @@ char * display_permission(unsigned int perms) {
 }
 
 char * display_guild_flags(unsigned int flags) {
-  char buf[256];
+  char buf[512];
```

Comment:
This is really important here. The longest possible output string is longer than 256 bytes.